pkg/controller/workloadspread/workloadspread_controller_utils.go

```diff
@@ -85,18 +85,53 @@ func filterOutCondition(conditions []appsv1alpha1.WorkloadSpreadSubsetCondition,
 	return newConditions
 }
 
-func matchesSubset(pod *corev1.Pod, node *corev1.Node, subset *appsv1alpha1.WorkloadSpreadSubset) (bool, int64, error) {
+func matchesSubset(pod *corev1.Pod, node *corev1.Node, subset *appsv1alpha1.WorkloadSpreadSubset, patch *corev1.Pod) (bool, int64, int64, error) {
 	matched, err := matchesSubsetRequiredAndToleration(pod, node, subset)
 	if err != nil || !matched {
-		return false, -1, err
+		return false, -1, -1, err
 	}
 
-	if len(subset.PreferredNodeSelectorTerms) == 0 {
-		return matched, 0, nil
+	// preferredNodeScore is in [0, total_prefer_weight]
+	preferredNodeScore := int64(0)
+	if subset.PreferredNodeSelectorTerms != nil {
+		nodePreferredTerms, _ := nodeaffinity.NewPreferredSchedulingTerms(subset.PreferredNodeSelectorTerms)
+		preferredNodeScore = nodePreferredTerms.Score(node)
 	}
 
-	nodePreferredTerms, _ := nodeaffinity.NewPreferredSchedulingTerms(subset.PreferredNodeSelectorTerms)
-	return matched, nodePreferredTerms.Score(node), nil
+	// preferredPodScore is in [0, 2]
+	preferredPodScore := int64(0)
+	if subset.Patch.Raw != nil {
+		preferredPodScore = podPreferredScore(subset, patch, pod)
+	}
+
+	return matched, preferredNodeScore, preferredPodScore, nil
+}
+
+func podPreferredScore(subset *appsv1alpha1.WorkloadSpreadSubset, patch, pod *corev1.Pod) int64 {
+	if subset.Patch.Raw == nil || patch == nil {
+		return 0
+	}
+	// return true if a contains b
+	isContained := func(a, b map[string]string) bool {
+		for k, v := range b {
+			if a[k] != v {
+				return false
+			}
+		}
+		return true
+	}
+
+	// currently, we only consider labels and annotations
```

Comment:
we can test for other field easily . We just clone the pod and apply the patch to the cloned pods, can compared the patched pod and original pod to see if they're identical